### ISSUE TITLE
Decouple from Fedora & dor-services

### ISSUE THREAD

**Issue by mjgiarlo** [sul-dlss/dor_indexing_app#588]
Based on current understanding, this ticket is likely blocked until [this production Argo query](https://argo.stanford.edu/catalog?f%5Bdata_quality_ssim%5D%5B%5D=Cocina+conversion+failed) returns 0 hits. Until that time, we need machinery in place to index objects not representable in Cocina by using the dor-services gem. Once that is done:

* Remove `DataQualityIndexer` entirely
* Replace `Dor.find` and Fedora-isms in `DorController`
* Replace multiple `Dor` model references (*e.g.*, `Dor::Agreement`) in `Indexer` with Cocina model references
* Remove `config/initializers/dor_config.rb` and, if needed, use another way to configure the Solr URL

**Comment by jcoyne** [comment]
@mjgiarlo the DataQualityIndexer is only used when we have no cocina representation available. (https://github.com/sul-dlss/dor_indexing_app/blob/a35431524359085bcdb159a35b027cb5bb1a7655/app/services/indexer.rb#L52-L59)   We can remove that entire class when all of our data is cocina compatible.  This is why we had tickets such as:

* https://github.com/sul-dlss/argo/issues/2460
* https://github.com/sul-dlss/argo/issues/2458

And more generally all the items returned by https://argo.stanford.edu/catalog?f[data_quality_ssim][]=Cocina+conversion+failed

**Comment by ndushay** [comment]
This *may* only have the rights/access indexing ticket left

**Comment by jcoyne** [comment]
I think we have to wait until we decommission Hydrus in order to finish this off.  Otherwise we have no way to index the invalid objects it creates into Argo.

There are also the EEMs objects that need to be dealt with: https://argo.stanford.edu/catalog?f[data_quality_ssim][]=Cocina+conversion+failed

**Comment by mjgiarlo** [comment]
@jcoyne Do  you reckon this issue is closeable now?

**Comment by jcoyne** [comment]
@mjgiarlo it's closeable by #807 which is on hold until migration

**Comment by mjgiarlo** [comment]
@jcoyne 💬 
> @mjgiarlo it's closeable by #807 which is on hold until migration

Excellent, thank you
